napari/utils/list/_model.py

```diff
@@ -28,38 +29,66 @@ def __init__(self, basetype, iterable=(), lookup=None):
         super().__init__(basetype, iterable, lookup)
         self.events = EmitterGroup(
             source=self,
-            auto_connect=True,
-            added=None,
-            removed=None,
-            reordered=None,
-            changed=None,
+            auto_connect=False,
+            added=Event,
+            removed=Event,
+            reordered=Event,
+            changed=Event,
         )
-        self.events.added.connect(self.events.changed)
-        self.events.removed.connect(self.events.changed)
-        self.events.reordered.connect(self.events.changed)
+        self.event_handler = EventHandler(component=self)
+        self.events.connect(self.event_handler.on_change)
 
     def __setitem__(self, query, values):
-        indices = tuple(self.__prsitem__(query))
-        new_indices = tuple(values)
-
-        if sorted(indices) != sorted(self.index(v) for v in new_indices):
-            raise TypeError(
-                'must be a reordering of indices; '
-                'setting of list items not allowed'
-            )
-
-        super().__setitem__(indices, new_indices)
-        self.events.reordered()
+        new_indices = tuple(self.__prsitem__(query))
+        old_indices = tuple(self.index(v) for v in tuple(values))
+        self.events.reordered((old_indices, new_indices))
+        self.events.changed(None)
 
     def insert(self, index, obj):
-        super().insert(index, obj)
-        self.events.added(item=obj, index=self.__locitem__(index))
+        self.events.added((obj, index))
+        self.events.changed(None)
 
     def append(self, obj):
-        TypedList.append(self, obj)
-        self.events.added(item=obj, index=len(self) - 1)
+        self.events.added((obj, len(self)))
+        self.events.changed(None)
 
     def pop(self, key):
-        obj = super().pop(key)
-        self.events.removed(item=obj, index=key)
+        obj = self[key]
+        self.events.removed((obj, key))
+        self.events.changed(None)
         return obj
+
+    def clear(self):
+        while len(self) > 0:
+            obj = self[0]
+            self.events.removed((obj, 0))
+        self.events.changed(None)
```

Comment:
The `changed` seems hacky... again, (a) can we rename to `any`, so that responding is `_on_any_change`, and (b), is there some way to somehow make all other events raise this automatically, rather than have to sprinkle `self.events.changed(None)`/`self.events.any(None)` everywhere?

Reply:
I'd like to drop the whole event during my next PR, but can we live with `changed` for one more, as it already existed before this? See how it got connected above.